### PR TITLE
PLAT-77054: Change components to allow disabled components to be focusable

### PR DESCRIPTION
**Pull request by webOS101**
Enact-DCO-1.0-Signed-off-by: Roy Sutton roy.sutton@lge.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [x] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Disabled spottable elements should be focusable.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Update `Spottable` to allow focus onto disabled items.  Disabled items themselves are still non-interactive but can have spotlight.  Updated focus CSS for many items as disabled + focused was not used before.  Also, I removed the `isItemDisabled` property from VL and VGL.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Design of some components may still be in flux so it is possible these designs will change further.  We don't have any specific guidelines for how disabled + focused should look, but in most cases it will be an opacity applied to the focused element.

I did not add a PR for deprecating the `isItemDisabled` property.

### Links
[//]: # (Related issues, references)
PLAT-77054

### CODE REVIEW

File: packages/spotlight/Spottable/Spottable.js [hunk 1]
```diff
@@ -393,7 +393,7 @@ const Spottable = hoc(defaultConfig, (config, Wrapped) => {
 
 		render () {
 			const {disabled, spotlightId, spotlightDisabled, ...rest} = this.props;
-			this.focusedWhenDisabled = this.isFocused && (disabled || spotlightDisabled);
+			this.focusedWhenDisabled = this.isFocused && spotlightDisabled;
```
Comment (viodragon2) [inline packages/spotlight/Spottable/Spottable.js]: Maybe we could unwind this and remove `forceUpdate` that happens in blur event.

Comment (webOS101) [reply packages/spotlight/Spottable/Spottable.js]: After re-reviewing, I realized this also handles the case where a component becomes `spotlightDisabled` when focused, which is not changed with the PR.  I think it should remain.

**Comment by ryanjduffy** [comment]
Few visual notes from sampler:
* Input - Field is white with full opacity
* ExpandableList/DatePicker/TimePicker - LabeledItem doesn't change
* MediaOverlay - Focus ring appears to be full opacity

**Comment by Djspaceg** [comment]
@ryanjduffy would you mind re-verifying that "ExpandableList/DatePicker/TimePicker - LabeledItem doesn't change"? I just checked again on my machine and it appears to be behaving correctly with yesterday's set of commits.

**Comment by ryanjduffy** [comment]
This is what I see with the latest. It shows the focused+disabled state okay but disabled+unfocused is the same as enabled+unfocused.

image

**Comment by Djspaceg** [comment]
Ah! Focused+notDisabled. Thanks for clarifying! I'll get that sorted out too.

**Comment by ryanjduffy** [comment]
One more visual note: FormCheckbox's text becomes white when focused but it doesn't change when disabled and focused which is inconsistent with other components that still change color but are "muted" by the opacity change.

**Comment by Djspaceg** [comment]
The above (previous 3) concerns were addressed, and a new knob was added to the MediaOverlay story.

I don't have a good answer for the FormCheckboxItem scenario. The white focus text is a recent change, and we've not received complete designs on exactly how those colors should be handled in every situation (like when it's disabled).

Several visual changes to FormCheckbox and Checkbox may be changing back to the 2.x style, which may affect this issue.

What do you want to do?

**Comment by ryanjduffy** [comment]
My opinion is to be consistent in the absence of specific designs but I will support whatever decision you make.

**Comment by ryanjduffy** [comment]
Also, please update the description of the PR to describe the changes.

File: packages/sampler/stories/default/MediaOverlay.js [hunk 2]
```diff
@@ -77,6 +77,7 @@ storiesOf('Moonstone', module)
 			const placeholder = prop.placeholder[placeholderName];
 			return (
 				<MediaOverlay
+					disabled={boolean('disabled', Config)}
```
Comment (webOS101) [inline packages/sampler/stories/default/MediaOverlay.js]: I'm not sure `MediaOverlay` is _supposed_ to be disable-able.  So, I'm not sure this makes sense.

Comment (Djspaceg) [reply packages/sampler/stories/default/MediaOverlay.js]: 🤷‍♂ I just did what I was told. I'm not familiar with how MediaOverlay is used.

**Comment by webOS101** [comment]
I was not able to observe the `VirtualList` disabled 5-way navigation issue.  It seems like it's up to the app to pass in a `isItemDisabled()` function so it's possible the app you were testing supplied this.  I updated the joined pickers and the PR description.

**Comment by ryanjduffy** [comment]
I should have been more clear. You're right that the app must provide that function but I don't think the function should be supported since its purpose was to assist in focus management.

**Comment by webOS101** [comment]
Hmm...  I guess we could still supply it for a `spotlightDisabled` sort of thing...  but...  I can remove it.

**Comment by webOS101** [comment]
Removed.

**Pull request merged**